sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/junit/KalixTestKitResource.java

```diff
@@ -60,7 +63,7 @@
  * }
  * </pre>
  */
-public final class KalixTestKitResource extends ExternalResource {
+public final class KalixTestKitResource extends ExternalResource implements BeforeAllCallback, AfterAllCallback {
```

Comment:
When implementing this two interfaces (`BeforeAllCallback` and `AfterAllCallback`), we can use this class as a Registered Extension. Junit will take care of the lifecycle for us. This is the equivalent of `ClassRule` in junit4.


`KalixTestKitResource` has already a hard dependency on junit4. In the sbt project, junit4 is marked as provided and then we add it to the pom explicitly. 

With this change, the `KalixTestKitResource` will also depend on junit5 and we will need both in the pom. That's problematic for existing projects. If a user update the SDK version, but doesn't touch the pom, they will get an error.

Therefore, as mentioned above, we can solve it by added junit libraries as transitive dependencies.

Reply:
Hmm, that could be an option. 

That will make it easier for us to deprecate junit4 support.

I will PR this.